library/src/main/java/org/akvo/flow/mapbox/offline/reactive/DeleteOfflineArea.kt

```diff
@@ -0,0 +1,40 @@
+package org.akvo.flow.mapbox.offline.reactive
+
+import android.content.Context
+import android.util.Log
+import com.mapbox.mapboxsdk.offline.OfflineRegion
+import io.reactivex.Completable
+
+class DeleteOfflineArea(private val context: Context) {
+
+    private val regionsLister = OfflineRegionsLister(context)
+
+    fun execute(regionId: Long): Completable {
+        return regionsLister.getRegionsList().flatMapCompletable { deleteArea(it, regionId) }
+    }
+
+    private fun deleteArea(regions: Array<OfflineRegion>, regionId: Long): Completable {
+        val region = regions.first { regionId == it.id }
```

Comment:
Don't understand this.

If we are iterating over all regions in line 13, why `region` here cannot be `null`? I was expecting that after getting the list of regions we would filter/find to find the correct region to delete.

Line 20 seems to assume that the `region` in line 17 will never be null.